File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.cpp
class Solution {
public:
    vector<int> rearrangeArray(vector<int>& nums) {
        int i = 0, j = 0, n = nums.size();
        vector<int>ans(n, 0);
        int k = 0;
        for(int i = 0; i < n; i++){
            if(nums[i] > 0){
                ans[k] = nums[i];
                k = k + 2;
            }
        }
        k = 1;
        for(int i = 0; i < n; i++){
            if(nums[i] < 0){
                ans[k] = nums[i];
                k = k + 2;
            }
        }
        return ans;
    }
};

File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.java
class Solution {
    public int[] rearrangeArray(int[] nums) {
        int n = nums.length;
        int[] res = new int[n];
        
        int i = 0, j = 0, k = 0;
        
        while(k < n){
            while(i < n && nums[i] < 0) ++i;
            if(i < n && nums[i] >= 0)   res[k++] = nums[i++];
        
            while(j < n && nums[j] >= 0) ++j;
                if(j < n && nums[j] < 0)    res[k++] = nums[j++];
        }
        
        return res;
    }
}